core/transcoder.go

```diff
@@ -31,14 +31,15 @@ func (lt *LocalTranscoder) Transcode(fname string, profiles []ffmpeg.VideoProfil
 	}
 	defer os.RemoveAll(fullDirName)
 	tr := transcoder.NewFFMpegSegmentTranscoder(profiles, fullDirName)
-	mid, seqNo, parseErr := parseURI(fname)
+	_, seqNo, parseErr := parseURI(fname)
 	start := time.Now()
-	if monitor.Enabled && parseErr == nil {
-		monitor.LogSegmentTranscodeStarting(seqNo, mid)
-	}
 	data, err := tr.Transcode(fname)
 	if monitor.Enabled && parseErr == nil {
-		monitor.LogSegmentTranscodeEnded(seqNo, mid, time.Since(start), common.ProfilesNames(profiles))
+		// This will run only when fname is actual URL and contains seqNo in it.
+		// When orchestrator works as transcoder, `fname` will be relative path to file in local
+		// filesystem and will not contain seqNo in it. For that case `SegmentTranscoded` will
+		// be called in orchestrator.go
+		monitor.SegmentTranscoded(0, seqNo, time.Since(start), common.ProfilesNames(profiles))
```

Comment:
It seems like we're tracking three events under this same metric:

* Orchestrator transcode complete
* Transcoder transcoded
* Broadcaster transcode complete

Distinguishing between those seems non-obvious. Would it make sense to track these separately?

Reply:
These are same events, but happening on different nodes. It is matter of query to distinguish between them:

`livepeer_segment_transcoded_total{node_type="orch"}`
or `livepeer_segment_transcoded_total{node_type="bctr"}`

Like [this](http://34.83.171.222:9090/graph?g0.range_input=1h&g0.expr=livepeer_segment_transcoded_total%7Bnode_type%3D%22orch%22%7D&g0.tab=1)